### PR TITLE
feat(motion_utils): add insertDecelPoint

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

add `insertDecelPoint` to motion_utils

## Tests performed

test code

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/motion_utils/include/motion_utils/trajectory/trajectory.hpp [hunk 1]
```diff
@@ -1245,6 +1245,46 @@ inline boost::optional<size_t> insertStopPoint(
...
+  }
+
+  const auto insertVelocity = [&insert_idx](T & points_with_twist, const float v) {
+    for (size_t i = insert_idx.get(); i < points_with_twist.size(); ++i) {
+      const auto & original_velocity =
+        tier4_autoware_utils::getLongitudinalVelocity(points_with_twist.at(i));
+      tier4_autoware_utils::setLongitudinalVelocity(
+        std::min(static_cast<float>(original_velocity), v), points_with_twist.at(i));
+    }
+  };
```
Comment (purewater0901) [inline common/motion_utils/include/motion_utils/trajectory/trajectory.hpp:1281]: Do you need to make this a function? I think this is not necessary

Comment (kosuke55) [reply common/motion_utils/include/motion_utils/trajectory/trajectory.hpp]: it's just copy from @satoshi-ota's function. but it seems no reason.
fixed in https://github.com/autowarefoundation/autoware.universe/pull/3226/commits/f0214b725b1a5bb2cfb05638f398efc9ddb1cc2c

**Review by purewater0901 — APPROVED** [review]
LGTM

File: common/motion_utils/include/motion_utils/trajectory/trajectory.hpp (hunk 1 above)
Comment (kosuke55) [reply common/motion_utils/include/motion_utils/trajectory/trajectory.hpp]: FYI: https://github.com/autowarefoundation/autoware.universe/blob/feat/insertDecelPoint/planning/behavior_path_planner/src/util/avoidance/util.cpp#L1000-L1007

**Pull request merged**